pkg/scheduler/plugins/loadaware/pod_assign_cache.go

```diff
@@ -211,3 +343,287 @@ func (p *podAssignCache) OnDelete(obj interface{}) {
 	}
 	p.unAssign(pod.Spec.NodeName, pod)
 }
+
+func (n *nodeInfo) AddOrUpdatePod(pod *podAssignInfo, locked bool) bool {
+	if locked {
```

Comment:
Recommended to simplify this implementation. We accept some redundant code to improve readability.